src/convert/slices.rs

```diff
@@ -271,3 +271,63 @@ if_std! {
         fn is_none(slice: &WasmSlice) -> bool { slice.ptr == 0 }
     }
 }
+
+
+macro_rules! array_impls {
+    ($($N:expr),+) => {
+        $(
+            impl<T> FromWasmAbi for [T; $N]
+            where
+                T: Copy + FromWasmAbi<Abi = WasmSlice>
+            {
+                type Abi = WasmSlice;
+        
+                #[inline]
+                unsafe fn from_abi(js: Self::Abi) -> Self {
+                    use core::convert::TryInto;
+                    let slice = slice::from_raw_parts(
+                        <*const T>::from_abi(js.ptr),
+                        js.len as usize,
+                    );
+                    slice.try_into().unwrap()
+                }
+            }
+
+            impl<T> IntoWasmAbi for [T; $N]
+            where
+                T: Copy + IntoWasmAbi<Abi = WasmSlice>
+            {
+                type Abi = WasmSlice;
+            
+                #[inline]
+                fn into_abi(self) -> Self::Abi {
+                    WasmSlice {
+                        ptr: self.as_ptr().into_abi(),
```

Comment:
I believe this is a use-after-free, right? `self` is dropped at the end of this function but we're returning a pointer to it?

Reply:
Hehehe. I should have paid more attention instead of simply copying and pasting. Thank you